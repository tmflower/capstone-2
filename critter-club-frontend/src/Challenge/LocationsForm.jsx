import { useState } from 'react';
import { Paper, FormLabel, Checkbox, Typography, Button, FormGroup } from '@mui/material';
import { theme } from '../theme/theme';

/** LocationsForm renders a question with checkbox answers */

export function LocationsForm({commonName, locations, message, points, setPoints, numQuestions, setNumQuestions, validLocations  }) {

    // set each location choice to unchecked by default
    const initialState = { 
        Africa: false,
        Antarctica: false,
        Asia: false,
        Europe: false,
        NorthAmerica: false,
        Eurasia: false,
        Ocean: false,
        CentralAmerica: false,
        SouthAmerica: false,
        Oceania: false
    }

    const [formData, setFormData] = useState(initialState);
    const [feedback, setFeedback] = useState('');

    // Deconstruct formData properties
    const {         
        Africa, 
        Antarctica, 
        Asia,
        Europe,
        NorthAmerica,
        Eurasia,
        Ocean,
        CentralAmerica,
        SouthAmerica,
        Oceania
    } = formData;

    // Set formData to user's input
    const handleChange = (e) => {
        const { name, checked } = e.target;
		setFormData((formData) => 
            ({ ...formData, [name]: checked }));
    }

    // narrow the list of valid locations to compare to user's choices
    // this prevents problems resulting from missing or messy location data from the api
//     const validLocations = [ 
//         "Africa",
//         "Antarctica",
//         "Asia",
//         "Europe",
//         "NorthAmerica",
//         "Eurasia",
//         "Ocean",
//         "CentralAmerica",
//         "SouthAmerica",
//         "Oceania"
//    ];

   // compare user's selection to animal data to check if correct; provide corresponding feedback message
    // update user's points
    // increment numQuestions; when numQuestions is >=3, user can submit answers and earn badge
   const handleSubmit = (e) => {
    e.preventDefault();
    // create an array of values from user input
    let userLocationChoices = [];
    for (let validLocation of validLocations) {
        if (formData[validLocation] === true) {
            userLocationChoices.push(validLocation);
        }
    }
    // compare length of user's array to animal data locations array
    // if not equal, there is no reason to check further; this also handles no user input
    // provide corresponding feedback message
    if (userLocationChoices.length !== locations.length) {
        setFeedback(message.incorrect);
    }
    // sort user's array to match the order of the animal data
    // compare values at matching index from each array
    // provide corresponding feedback message
    userLocationChoices.sort();
    console.log("userLocationChoices:", userLocationChoices);
    console.log("locations:", locations)
    let pointer1 = 0;
    let pointer2 = 0;

    if (userLocationChoices.length === 1 &&
        locations.length === 1 &&
        userLocationChoices[0] === locations[0]) {
            setFeedback(message.correct); 
            setPoints(points+=10);
            setNumQuestions(numQuestions+1);
        }
    else {

        while (pointer1 < userLocationChoices.length) {            
            if (userLocationChoices[pointer1] !== locations[pointer2]) {
                setFeedback(message.incorrect);
                break;
            }
                else {
                    pointer1++;
                    pointer2++;
                        if (pointer1 === userLocationChoices.length -1 && 
                            pointer2 === locations.length -1 && 
                            userLocationChoices[pointer1] === locations[pointer2]) {
                                setFeedback(message.correct); 
                                setPoints(points+=10);
                                setNumQuestions(numQuestions+1);
                        }
                }
            }
        }        
    }
    
    // clear form responses & message when user clicks "Try again" button
    const handleReset = (e) => {
        setFormData(initialState);
        setFeedback('');
    }

    return (
        <Paper
        elevation={8}
        sx={{ padding: 3, mt: 3 }}>
        <form onSubmit={handleSubmit}>
                {validLocations.includes(locations[0]) ? 
                <div>
                    <fieldset>
                    <FormGroup>
                    <FormLabel sx={{ fontSize: '1.5rem', fontFamily: 'Lexend Deca, Arial',}}>
                        <legend>In which location or locations can the {commonName.toLowerCase()} be found?
                        </legend>
                    </FormLabel>
                        <label htmlFor='Africa'>
                            <Checkbox
                                // type="checkbox"
                                id="Africa"
                                name="Africa"
                                checked={Africa}
                                onChange={handleChange}
                                sx={{ '& .MuiSvgIcon-root': { fontSize: 28 } }}
                            />
                            Africa
                        </label>
                        <label htmlFor='Antarctica'>
                            <Checkbox
                                // type="checkbox"
                                id="Antarctica"
                                name="Antarctica"
                                checked={Antarctica}
                                onChange={handleChange}
                            />
                            Antarctica
                        </label>
                        <label htmlFor='Asia'>
                            <Checkbox
                                // type="checkbox"
                                id="Asia"
                                name="Asia"
                                checked={Asia}
                                onChange={handleChange}
                            />
                            Asia
                        </label>
                        <label htmlFor='Europe'>
                            <Checkbox
                                // type="checkbox"
                                id="Europe"
                                name="Europe"
                                checked={Europe}
                                onChange={handleChange}
                            />
                            Europe
                        </label>
                        <label htmlFor='North America'>
                            <Checkbox
                                // type="checkbox"
                                id="North America"
                                name="NorthAmerica"
                                checked={NorthAmerica}
                                onChange={handleChange}
                            />
                            North America
                        </label>
                        <label htmlFor='Eurasia'>
                            <Checkbox
                                // type="checkbox"
                                id="Eurasia"
                                name="Eurasia"
                                checked={Eurasia}
                                onChange={handleChange}
                            />
                            Eurasia
                        </label>
                        <label htmlFor='Ocean'>
                            <Checkbox
                                // type="checkbox"
                                id="Ocean"
                                name="Ocean"
                                checked={Ocean}
                                onChange={handleChange}
                            />
                            Ocean
                        </label>
                        <label htmlFor='Central America'>
                            <Checkbox
                                // type="checkbox"
                                id="Central America"
                                name="CentralAmerica"
                                checked={CentralAmerica}
                                onChange={handleChange}
                            ></Checkbox>
                            Central America
                        </label>
                        <label htmlFor='South America'>
                            <Checkbox
                                // type="checkbox"
                                id="South America"
                                name="SouthAmerica"
                                checked={SouthAmerica}
                                onChange={handleChange}
                            />
                            South America
                        </label>
                        <label htmlFor='Oceania'>
                            <Checkbox
                                // type="checkbox"
                                id="Oceania"
                                name="Oceania"
                                checked={Oceania}
                                onChange={handleChange}
                            />
                            Oceania
                        </label>
                        <div>
                            { !feedback ? 
                            <Button type="submit">Check answers</Button>
                            : null }
                            { feedback === message.incorrect ?
                            <Button type="reset" onClick={handleReset}>Clear answers</Button>
                            : null }
                        </div>
                        <Typography id="quiz-feedback" variant="h5" sx={{ fontFamily: theme.typography.primary, textAlign: 'center' }}>{feedback}</Typography>  
                    </FormGroup>                      
                    </fieldset>                   
                </div>                 
                : null}
        </form>
        </Paper>
    )
}