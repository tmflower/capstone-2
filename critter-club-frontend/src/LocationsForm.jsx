import { useState } from 'react';

export function LocationsForm({commonName, locations}) {

    // set each location choice to unchecked by default
    const initialState = { 
        Africa: false,
        Antarctica: false,
        Asia: false,
        Europe: false,
        NorthAmerica: false,
        Eurasia: false,
        Ocean: false,
        CentralAmerica: false,
        SouthAmerica: false,
        Oceania: false
    }

    const [formData, setFormData] = useState(initialState);

    const {         
        Africa, 
        Antarctica, 
        Asia,
        Europe,
        NorthAmerica,
        Eurasia,
        Ocean,
        CentralAmerica,
        SouthAmerica,
        Oceania
    } = formData;

    const handleChange = (e) => {
        const { name, checked } = e.target;
		setFormData((formData) => 
            ({ ...formData, [name]: checked }));
    }

    // narrow the list of valid locations to compare to user's choices
    // this prevents problems resulting from missing or messy location data from the api
    const validLocations = [ 
        "Africa",
        "Antarctica",
        "Asia",
        "Europe",
        "NorthAmerica",
        "Eurasia",
        "Ocean",
        "CentralAmerica",
        "SouthAmerica",
        "Oceania"
   ];

   const handleSubmit = (e) => {
    // TODO: add functionality for user feedback
    e.preventDefault();
    // create an array of values from user input
    let userLocationChoices = [];
    for (let validLocation of validLocations) {
        if (formData[validLocation] === true) {
            userLocationChoices.push(validLocation);
        }
    }
    // compare length of user's array to animal data locations array
    // if not equal, there is no reason to check further; this also handles no user input
    if (userLocationChoices.length !== locations.length) {
        console.log("No on locations--length!!! ðŸ˜«ðŸ˜«ðŸ˜«")
    }
    // sort user's array to match the order of the animal data
    // compare values at matching index from each array
    userLocationChoices.sort();
    console.log("userLocationChoices:", userLocationChoices);
    console.log("locations:", locations)
    let pointer1 = 0;
    let pointer2 = 0;
    while (pointer1 < userLocationChoices.length) {            
        if (userLocationChoices[pointer1] !== locations[pointer2]) {
            console.log("No on locations!!! ðŸ˜«ðŸ˜«ðŸ˜«");
            break;
        }
        else if (userLocationChoices[pointer1] === locations[pointer2]) {
            pointer1++;
            pointer2++;
            if (pointer1 === userLocationChoices.length -1 && 
                pointer2 === locations.length -1 && 
                userLocationChoices[pointer1] === locations[pointer2]) {
                console.log("YOU ARE CORRECT---locations!!! ðŸ˜€ðŸ˜€ðŸ˜€") 
            }
        }
        }        
    }
    const handleReset = (e) => {
        setFormData(initialState);
    }

    return (
        <form onSubmit={handleSubmit}>
                {validLocations.includes(locations[0]) ? 
                <div>
                    <fieldset>
                        <legend>In which location or locations can {commonName.toLowerCase()} be found?
                        </legend>
                        <label>
                            <input
                                type="checkbox"
                                id="Africa"
                                name="Africa"
                                checked={Africa}
                                onChange={handleChange}
                            ></input>
                            Africa
                        </label>
                        <label>
                            <input
                                type="checkbox"
                                id="Antarctica"
                                name="Antarctica"
                                checked={Antarctica}
                                onChange={handleChange}
                            ></input>
                            Antarctica
                        </label>
                        <label>
                            <input
                                type="checkbox"
                                id="Asia"
                                name="Asia"
                                checked={Asia}
                                onChange={handleChange}
                            ></input>
                            Asia
                        </label>
                        <label>
                            <input
                                type="checkbox"
                                id="Europe"
                                name="Europe"
                                checked={Europe}
                                onChange={handleChange}
                            ></input>
                            Europe
                        </label>
                        <label>
                            <input
                                type="checkbox"
                                id="NorthAmerica"
                                name="NorthAmerica"
                                checked={NorthAmerica}
                                onChange={handleChange}
                            ></input>
                            North America
                        </label>
                        <label>
                            <input
                                type="checkbox"
                                id="Eurasia"
                                name="Eurasia"
                                checked={Eurasia}
                                onChange={handleChange}
                            ></input>
                            Eurasia
                        </label>
                        <label>
                            <input
                                type="checkbox"
                                id="Ocean"
                                name="Ocean"
                                checked={Ocean}
                                onChange={handleChange}
                            ></input>
                            Ocean
                        </label>
                        <label>
                            <input
                                type="checkbox"
                                id="CentralAmerica"
                                name="CentralAmerica"
                                checked={CentralAmerica}
                                onChange={handleChange}
                            ></input>
                            Central America
                        </label>
                        <label>
                            <input
                                type="checkbox"
                                id="SouthAmerica"
                                name="SouthAmerica"
                                checked={SouthAmerica}
                                onChange={handleChange}
                            ></input>
                            South America
                        </label>
                        <label>
                            <input
                                type="checkbox"
                                id="Oceania"
                                name="Oceania"
                                checked={Oceania}
                                onChange={handleChange}
                            ></input>
                            Oceania
                        </label>
                        <div>
                            <button type="submit">Check answer</button>
                            <button type="reset" onClick={handleReset}>Clear answer</button>
                        </div>                        
                    </fieldset>                   
                </div>                 
                : null}
        </form>
    )
}