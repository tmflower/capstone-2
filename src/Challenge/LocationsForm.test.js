import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { LocationsForm } from './LocationsForm';
import UserContext from '../userContext';
import userEvent from "@testing-library/user-event";


const currentUser = {user: { username: "Tiny", id: 1, password: 'abc123$%^', points: 40, userBadges: ['Koala', 'Jellyfish', 'Painted Turtle'], parentId: 1 }}

const message = {
  correct: "⭐️⭐️⭐️ You got it! ⭐️⭐️⭐️",
  incorrect: "😧😧😧 Sorry, try again. 😧😧😧"
};

const locations = ["Africa"];

const validLocations = [ 
  "Africa",
  "Antarctica",
  "Asia",
  "Europe",
  "NorthAmerica",
  "Eurasia",
  "Ocean",
  "CentralAmerica",
  "SouthAmerica",
  "Oceania"
];

// smoke test
test("renders without crashing", () => {

    render(
    <MemoryRouter>
      <UserContext.Provider value={currentUser}>
        <LocationsForm 
          commonName="Hippopotamus"
          message={message}
          locations={locations}
          validLocations={validLocations}/>
      </UserContext.Provider>
    </MemoryRouter>);
  });

// snapshot test
test("matches snapshot", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <UserContext.Provider value={currentUser}>
          <LocationsForm 
            commonName="Hippopotamus"
            message={message}
            locations={locations}
            validLocations={validLocations}/>
        </UserContext.Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

// tests for user interactions
test("accepts user's selected answer and responds with feedback message", async () => {

  const user = userEvent.setup();

  render(  
    <MemoryRouter>
      <UserContext.Provider value={currentUser}>
        <LocationsForm 
        commonName="Hippopotamus"
        message={message}
        locations={locations}
        validLocations={validLocations}
        />
      </UserContext.Provider>      
    </MemoryRouter>);

  const wrong = screen.getByLabelText("Antarctica");
  expect(wrong).not.toBeChecked();
  await user.click(wrong);
  expect(wrong).toBeChecked();
  // await user.click(screen.getByText("Check answer"));
  // expect(screen.getByText("😧😧😧 Sorry, try again. 😧😧😧")).toBeInTheDocument(); 

  const right = screen.getByLabelText("Africa");
  expect(right).not.toBeChecked();
  await user.click(right);
  expect(right).toBeChecked();
  // await user.click(screen.getByText("Check answer"));
  // expect(screen.getByText("⭐️⭐️⭐️ You got it! ⭐️⭐️⭐️")).toBeInTheDocument(); 
});