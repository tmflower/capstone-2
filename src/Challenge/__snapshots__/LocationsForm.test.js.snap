// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 css-2e4xct-MuiPaper-root"
  >
    <form>
      <div>
        <fieldset>
          <div
            class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary css-tshmje-MuiFormLabel-root"
            >
              <legend>
                In which location or locations can the hippopotamus be found?
              </legend>
            </label>
            <label
              for="Africa"
            >
              <span
                class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-1oakzvg-MuiButtonBase-root-MuiCheckbox-root"
              >
                <input
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  data-indeterminate="false"
                  id="Africa"
                  name="Africa"
                  type="checkbox"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CheckBoxOutlineBlankIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              Africa
            </label>
            <label
              for="Antarctica"
            >
              <span
                class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-1oakzvg-MuiButtonBase-root-MuiCheckbox-root"
              >
                <input
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  data-indeterminate="false"
                  id="Antarctica"
                  name="Antarctica"
                  type="checkbox"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CheckBoxOutlineBlankIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              Antarctica
            </label>
            <label
              for="Asia"
            >
              <span
                class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-1oakzvg-MuiButtonBase-root-MuiCheckbox-root"
              >
                <input
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  data-indeterminate="false"
                  id="Asia"
                  name="Asia"
                  type="checkbox"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CheckBoxOutlineBlankIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              Asia
            </label>
            <label
              for="Europe"
            >
              <span
                class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-1oakzvg-MuiButtonBase-root-MuiCheckbox-root"
              >
                <input
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  data-indeterminate="false"
                  id="Europe"
                  name="Europe"
                  type="checkbox"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CheckBoxOutlineBlankIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              Europe
            </label>
            <label
              for="North America"
            >
              <span
                class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-1oakzvg-MuiButtonBase-root-MuiCheckbox-root"
              >
                <input
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  data-indeterminate="false"
                  id="North America"
                  name="NorthAmerica"
                  type="checkbox"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CheckBoxOutlineBlankIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              North America
            </label>
            <label
              for="Eurasia"
            >
              <span
                class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-1oakzvg-MuiButtonBase-root-MuiCheckbox-root"
              >
                <input
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  data-indeterminate="false"
                  id="Eurasia"
                  name="Eurasia"
                  type="checkbox"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CheckBoxOutlineBlankIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              Eurasia
            </label>
            <label
              for="Ocean"
            >
              <span
                class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-1oakzvg-MuiButtonBase-root-MuiCheckbox-root"
              >
                <input
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  data-indeterminate="false"
                  id="Ocean"
                  name="Ocean"
                  type="checkbox"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CheckBoxOutlineBlankIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              Ocean
            </label>
            <label
              for="Central America"
            >
              <span
                class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-1oakzvg-MuiButtonBase-root-MuiCheckbox-root"
              >
                <input
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  data-indeterminate="false"
                  id="Central America"
                  name="CentralAmerica"
                  type="checkbox"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CheckBoxOutlineBlankIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              Central America
            </label>
            <label
              for="South America"
            >
              <span
                class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-1oakzvg-MuiButtonBase-root-MuiCheckbox-root"
              >
                <input
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  data-indeterminate="false"
                  id="South America"
                  name="SouthAmerica"
                  type="checkbox"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CheckBoxOutlineBlankIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              South America
            </label>
            <label
              for="Oceania"
            >
              <span
                class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-1oakzvg-MuiButtonBase-root-MuiCheckbox-root"
              >
                <input
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  data-indeterminate="false"
                  id="Oceania"
                  name="Oceania"
                  type="checkbox"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CheckBoxOutlineBlankIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              Oceania
            </label>
            <div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="submit"
              >
                Check answers
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <h5
              class="MuiTypography-root MuiTypography-h5 css-18czui-MuiTypography-root"
              id="quiz-feedback"
            />
          </div>
        </fieldset>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
